docs/ARCHITECTURE.md

```diff
@@ -240,24 +344,41 @@ direction LR
             +usingOutputMime(string $mimeType) self
             +usingOutputSchema(array< string, mixed > $schema) self
             +usingOutputModalities(...AiModality $modalities) self
+            +asArrayResponse(?array< string, mixed > $schema) self
+            +asJsonResponse(?array< string, mixed > $schema) self
             +generateResult() GenerativeAiResult
+            +generateResults(int $candidateCount) GenerativeAiResult[]
             +generateOperation() GenerativeAiOperation
+            +generateOperations(int $candidateCount) GenerativeAiOperation[]
             +generateTextResult() GenerativeAiResult
+            +generateTextResults(int $candidateCount) GenerativeAiResult[]
             +streamGenerateTextResult() Generator< GenerativeAiResult >
             +generateImageResult() GenerativeAiResult
+            +generateImageResults(int $candidateCount) GenerativeAiResult[]
             +convertTextToSpeechResult() GenerativeAiResult
+            +convertTextToSpeechResults(int $candidateCount) GenerativeAiResult[]
             +generateSpeechResult() GenerativeAiResult
+            +generateSpeechResults(int $candidateCount) GenerativeAiResult[]
             +generateEmbeddingsResult() EmbeddingResult
+            +generateEmbeddingsResults(int $candidateCount) EmbeddingResult[]
             +generateTextOperation() GenerativeAiOperation
+            +generateTextOperations(int $candidateCount) GenerativeAiOperation[]
             +generateImageOperation() GenerativeAiOperation
+            +generateImageOperations(int $candidateCount) GenerativeAiOperation[]
             +convertTextToSpeechOperation() GenerativeAiOperation
             +generateSpeechOperation() GenerativeAiOperation
+            +generateSpeechOperations(int $candidateCount) GenerativeAiOperation[]
             +generateEmbeddingsOperation() EmbeddingOperation
+            +generateEmbeddingsOperations(int $candidateCount) EmbeddingOperation[]
```

Comment:
Per my previous comment on the `generateTexts(4)` example, please keep in mind that the result will _always_ be a single `GenerativeAiResult` or a single `GenerativeAiOperation`. It's the _candidates_ within these objects that would be affected by the number of how many candidates you request.

In other words, all these `*Results()` and `*Operations()` methods would be invalid.

For the short-hand methods below (like `generateTexts()`) I like your idea though. Since the return value there is always the fully parsed result, having a plural short-hand to not have to separately call `usingCandidateCount(4)` makes a lot of sense to me.

Reply:
Ohhh, that makes sense. That was a fundamental misunderstanding on my part how the `usingCandidateCount(4)` would behave. I'll update the return types to reflect the actual return type as you mentioned above.